third_party/move/move-vm/runtime/src/reentrancy_checker.rs

```diff
@@ -155,10 +162,14 @@ impl ReentrancyChecker {
         }
     }
 
-    pub fn check_resource_access(&self, struct_id: &StructIdentifier) -> PartialVMResult<()> {
+    pub fn check_resource_access(
+        &self,
+        struct_id: &StructIdentifier,
+        module_hash: [u8; 32],
+    ) -> PartialVMResult<()> {
         if self
             .active_modules
-            .get(&struct_id.module)
+            .get(&module_hash)
```

Comment:
So on hash collision, the worst case is that we not allow access to resource, right, and blake3 is deterministic?

Reply:
Correct. Based on my limited understanding, blake3 is a cryptographic hash -- secure against collisions. In the near-impossible event of a collision, we incorrectly raise an error, causing the transaction to fail, which is not the end of the world.